onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -31,11 +33,11 @@ ONNX_OPERATOR_KERNEL_EX(
 
 MKLDNNExecutionProvider::MKLDNNExecutionProvider(const MKLDNNExecutionProviderInfo& /*info*/) {
   DeviceAllocatorRegistrationInfo default_allocator_info({OrtMemTypeDefault,
-                                                          [](int) { return std::make_unique<MKLDNNAllocator>(); }, std::numeric_limits<size_t>::max()});
+                                                          [](int) { return std::make_unique<CPUAllocator>(MKLDNN, OrtAllocatorType::OrtDeviceAllocator, 0, OrtMemTypeDefault); }, std::numeric_limits<size_t>::max()});
   InsertAllocator(CreateAllocator(default_allocator_info));
```

Comment:
Why do we need to different allocators if they're both CPUAllocator based?

Reply:
I'm in the way of figuring this out with this kind of refactoring...

Reply:
OK. What's the intention with this PR? Check it in with two allocators and further cleanups in separate PR/s, or you're going to add to this PR and this isn't the final set of changes yet?

---
In reply to: [255732643](https://github.com/Microsoft/onnxruntime/pull/467#discussion_r255732643) [](ancestors = 255732643)

Reply:
I will have this as the 1st PR to cleanup/refactor the allocator design.

Reply:
Is there any MKLDNN kernel that need to specify its input/output memory type is CPU memory? that looks weird...
I guess they are using this trick to avoid invoke "CopyTensor".